lib/osf-components/addon/components/file-actions-menu/submit-to-boa-modal/component.ts

```diff
@@ -0,0 +1,102 @@
+import { inject as service } from '@ember/service';
+import { waitFor } from '@ember/test-waiters';
+import Component from '@glimmer/component';
+import { task } from 'ember-concurrency';
+import IntlService from 'ember-intl/services/intl';
+import File from 'ember-osf-web/packages/files/file';
+import captureException, { getApiErrorMessage } from 'ember-osf-web/utils/capture-exception';
+import config from 'ember-osf-web/config/environment';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+interface Args {
+    file: File;
+    isOpen: boolean;
+    closeModal: () => {};
+}
+
+export default class SubmitToBoaModal extends Component<Args> {
+    @service toast!: Toastr;
+    @service intl!: IntlService;
+    datasets?: string[];
+    @tracked selectedDataset?: string;
+
+    datasets = [
+        '2022 Jan/Java',
+        '2022 Feb/Python',
+        '2021 Method Chains',
+        '2021 Aug/Python',
+        '2021 Aug/Kotlin (small)',
+        '2021 Aug/Kotlin',
+        '2021 Jan/ML-Verse',
+        '2020 August/Python-DS',
+        '2019 October/GitHub (small)',
+        '2019 October/GitHub (medium)',
+        '2019 October/GitHub',
+        '2015 September/GitHub',
+        '2013 September/SF (small)',
+        '2013 September/SF (medium)',
+        '2013 September/SF',
+        '2013 May/SF',
+        '2013 February/SF',
+        '2012 July/SF',
+    ];
```

Comment:
This is less than ideal, but I see it's been discussed already as part of the scope. Let's never update this list, and if it ever needs to change, we should use an endpoint to get the current list of datasets.